src/headingengine.js

```diff
@@ -75,17 +103,44 @@ export default class HeadingEngine extends Plugin {
 		const editor = this.editor;
 		const command = editor.commands.get( 'heading' );
 		const enterCommand = editor.commands.get( 'enter' );
+		const formats = this._formats;
 
 		if ( enterCommand ) {
 			this.listenTo( enterCommand, 'afterExecute', ( evt, data ) => {
 				const positionParent = editor.document.selection.getFirstPosition().parent;
 				const batch = data.batch;
-				const isHeading = formats.some( ( format ) => format.id == positionParent.name );
+				const isHeading = formats.some( ( { id } ) => id == positionParent.name );
 
 				if ( isHeading && positionParent.name != command.defaultFormat.id && positionParent.childCount === 0 ) {
 					batch.rename( positionParent, command.defaultFormat.id );
 				}
 			} );
 		}
 	}
+
+	/**
+	 * A set of formats as defined in `config.heading.formats`.
+	 *
+	 * @readonly
+	 * @protected
+	 * @type {Array.<module:heading/headingcommand~HeadingFormat>}
+	 */
+	get _formats() {
```

Comment:
OTOH, this more deserves to be a method – `_getLocalizedFormats()`. Plus, even the API doc is misleading now because it doesn't explain why this getter exists in the first place.